src/engine/engine.h

```diff
@@ -6,4 +6,28 @@
 namespace mixxx {
     // TODO(XXX): When we move from stereo to multi-channel this needs updating.
     static constexpr mixxx::AudioSignal::ChannelCount kEngineChannelCount(2);
+
+    // Contains the information needed to process a buffer of audio
+    class AudioParameters : public AudioSignal {
```

Comment:
I'm not happy with my decision to make AudioSignal a base class. Inheritance is only need for the polymorphic verifyReadable() function that you don't use. You can reduce the coupling with AudioSignal by making it a regular member which is sufficient here:
```
AudioSignal m_audioSignal;
SINT m_frameBufferLength;
```
I also like the frameBufferLength()/sampleBufferLength() wording which resembles the wording in AudioSource. Just choose what works best for you ;)

Reply:
I do not like the names "frameBufferLength" or "sampleBufferLength" because they imply that there are two different buffers, one for frames and one for samples. I prefer "samplesPerBuffer" and "framesPerBuffer".